ssz-rs/src/vector.rs

```diff
@@ -118,12 +118,11 @@ where
     T: SimpleSerialize + Default + Clone,
 {
     fn default() -> Self {
+        assert!(N > 0);
+
         let data = vec![T::default(); N];
-        match data.try_into() {
```

Comment:
I'm happy with either way too. The panic case can't come up anyway because `data.len() == N`. Removing the panic case makes this clearer when scanning the code, but then we lose the elegance of using `try_from` everywhere.

Reply:
If we keep `try_from`, we can shorten it with `map_err`: `data.try_into().map_err(|(_, err)| err).unwrap()`